File: server/routes/bot.js
import express from 'express';
import { requireAdmin } from '../middleware/auth.js';
import instagramBot from '../services/instagramBot.js';

const router = express.Router();

router.post('/send-update', requireAdmin, async (req, res) => {
  try {
    const { userId, content } = req.body;

    if (!userId || !content) {
      return res.status(400).json({ error: 'User ID and content are required' });
    }

    await instagramBot.sendEventUpdate(userId, content);
    res.json({ success: true });
  } catch (error) {
    res.status(500).json({ error: 'Failed to send update' });
  }
});

router.post('/broadcast', requireAdmin, async (req, res) => {
  try {
    const { content } = req.body;

    if (!content) {
      return res.status(400).json({ error: 'Content is required' });
    }

    const results = await instagramBot.broadcastUpdate(content);
    res.json({ success: true, results });
  } catch (error) {
    res.status(500).json({ error: 'Failed to broadcast update' });
  }
});

export default router;

File: app/routes/bot.py
from flask import Blueprint, request, jsonify
from app.middleware.auth import require_admin
from app.services.instagram_bot import InstagramBot

bot_bp = Blueprint('bot', __name__, url_prefix='/api/bot')
bot = InstagramBot()

@bot_bp.route('/send-update', methods=['POST'])
@require_admin
def send_update():
    data = request.get_json()
    user_id = data.get('user_id')
    content = data.get('content')
    
    if not user_id or not content:
        return jsonify({'error': 'User ID and content are required'}), 400
    
    try:
        success = bot.send_event_update(user_id, content)
        if success:
            return jsonify({'success': True})
        return jsonify({'error': 'Failed to send update'}), 500
    except Exception as e:
        return jsonify({'error': str(e)}), 500

@bot_bp.route('/broadcast', methods=['POST'])
@require_admin
def broadcast():
    data = request.get_json()
    content = data.get('content')
    
    if not content:
        return jsonify({'error': 'Content is required'}), 400
    
    try:
        results = bot.broadcast_update(content)
        return jsonify({'success': True, 'results': results})
    except Exception as e:
        return jsonify({'error': str(e)}), 500
